### DISCUSSION TITLE
Dynamically add otel javaagent to application at runtime

### DISCUSSION THREAD

**Discussion by sarkisjad** [open-telemetry/opentelemetry-java-instrumentation#9594] [Q&A]
Is it possible to dynamically add otel [javaagent](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/c7617dc06800c15ef5fbe729688c3fcbe2c1c0d4/instrumentation/jmx-metrics/javaagent/README.md) to an application at runtime?

**Comment by jeanbisutti** [comment]
@sarkisjad You can use [this project](https://github.com/open-telemetry/opentelemetry-java-contrib/tree/main/runtime-attach).

**Reply by sarkisjad** [reply]
Is it applicable for all java application? Even if they aren't SpringBoot?

**Reply by jeanbisutti** [reply]
It should work for applications not based on Spring Boot. The important thing is to call `RuntimeAttach.attachJavaagentToCurrentJVM()` at the beginning of the main method.

**Comment by jkwatson** [comment]
moving to the instrumentation project, which concerns the agent...
